ckanext/ckanext-apply_permissions_for_service/ckanext/apply_permissions_for_service/logic/auth.py

```diff
@@ -10,10 +10,12 @@ def service_permission_application_show(context, data_dict):
 
     permission_application_id = toolkit.get_or_bust(data_dict, 'id')
     application = model.ApplyPermission.get(permission_application_id).as_dict()
-    organization_id = application.get('organization')
+    organization = application.get('organization')
+    target_organization = application.get('target_organization')
     membership_organizations = toolkit.get_action('organization_list_for_user')(context, {'permission': 'read'})
 
-    if organization_id in [org.get('id') for org in membership_organizations]:
+    if any(True for x in [org.get('id') for org in membership_organizations]
```

Comment:
The inner comprehension here can be a generator instead of a list:
```suggestion
    if any(True for x in (org.get('id') for org in membership_organizations)
```